### ISSUE TITLE
Users can not update VM config

### ISSUE THREAD

**Issue by sergiojvg** [OpenNebula/one#5096]
**Description**
When a regular user tries to update the VM's config (for example, a context var), it gets the following error:

```
one.vm.updateconf result FAILURE [one.vm.updateconf] User [2] : Template includes a restricted attribute DISK.
```

The specific setting from oned.conf creating the issue is:

```
VM_RESTRICTED_ATTR = "DISK/ORIGINAL_SIZE"
```

The weird part is that this setting exists also in older versions (for example, 5.10.5), but this issue is non existent on that version.

**To Reproduce**
Create a user (not belonging to oneadmin group), allow this user to use a specific vm (use permission, change owner, whatever). Try to update the VM's context.

**Details**
 - Affected components: core
 - Version: 5.12.3

**Additional context**

## Progress Status
- [ ] Branch created 
- [ ] Code committed to development branch
- [ ] Testing - QA
- [ ] Documentation 
- [ ] Release notes - resolved issues, compatibility, known issues
- [ ] Code committed to upstream release/hotfix branches
- [ ] Documentation committed to upstream release/hotfix branches

**Comment by dfelsch** [comment]
I also just stumbled upon this. Is there any workaround? Removing `DISK/ORIGINAL_SIZE` from the `VM_RESTRICTED_ATTR` list does not seem to be a good idea as this value directly ends in shell scripts and would possibly allow RCE, correct?

**Comment by al3xhh** [comment]
Hello @sergiojvg,

Thank you for reporting the bug, I have tried to reproduce doing the following steps:

1. Create a normal user:

```
$ oneuser create test test

$ oneuser list --no-expand
  ID NAME            ENAB GROUP    AUTH            VMS     MEMORY        CPU
   2 test            yes  users    core        1 /   -    128M /   1.0 /   -
```

2. Instantiate a VM template with this user:

```
$ onetemplate instantiate 0 --user test
$ onevm list --no-expand
  ID USER     GROUP    NAME            STAT  CPU     MEM HOST             TIME
   0 test     users    Alpine Linux 3. runn    1    128M localhost    0d 00h05
```

3. Update the CONTEXT section using the command `onevm updateconf`:

```
$ onevm updateconf 0 --user test
$ echo $?
0
```

As you can see, I got no errors and the template is correctly updated.

4. Check that I have the VM_RESTRICTED_ATTRIBUTE:

```
$ grep "VM_RESTRICTED_ATTR = \"DISK/ORIGINAL_SIZE\"" /etc/one/oned.conf 
VM_RESTRICTED_ATTR = "DISK/ORIGINAL_SIZE"
```

So, could you please provide more information so I can try to reproduce it?

Best,
Álex.

**Comment by al3xhh** [comment]
I have already tried instantiating the VM as oneadmin and then sharing the VM with the user and it also worked.

**Comment by al3xhh** [comment]
Closing this until more information is provied.

**Comment by dfelsch** [comment]
Have you checked, that your VM's config contains a `DISK/ORIGINAL_SIZE` setting? Here's the section from a VM that I can reproduce the bug with:
```
DISK = [
  ALLOW_ORPHANS = "NO",
  CLONE = "YES",
  CLONE_TARGET = "SYSTEM",
  CLUSTER_ID = "100",
  DATASTORE = "default",
  DATASTORE_ID = "1",
  DEV_PREFIX = "vd",
  DISK_ID = "0",
  DISK_SNAPSHOT_TOTAL_SIZE = "0",
  DISK_TYPE = "FILE",
  DRIVER = "qcow2",
  IMAGE = "Ubuntu 20.04 LTS Server x64 - Kernel 5.4.0-37",
  IMAGE_ID = "353",
  IMAGE_STATE = "2",
  IMAGE_UNAME = "oneadmin",
  LN_TARGET = "NONE",
  ORDER = "1",
  ORIGINAL_SIZE = "8192",
  READONLY = "NO",
  SAVE = "NO",
  SIZE = "8192",
  SOURCE = "/var/lib/one//datastores/1/6002e2f76fe8ff2fd5c84a96093d12d1",
  TARGET = "vda",
  TM_MAD = "qcow2",
  TYPE = "FILE" ]
```

**Comment by al3xhh** [comment]
Hello @dfelsch,

Yes, my VM has that in the template:

```
DISK = [
  ALLOW_ORPHANS = "YES",
  CLONE = "YES",
  CLONE_TARGET = "SYSTEM",
  CLUSTER_ID = "0",
  DATASTORE = "default",
  DATASTORE_ID = "1",
  DEV_PREFIX = "vd",
  DISK_ID = "0",
  DISK_SNAPSHOT_TOTAL_SIZE = "0",
  DISK_TYPE = "FILE",
  DRIVER = "qcow2",
  FORMAT = "qcow2",
  IMAGE = "Alpine Linux 3.11",
  IMAGE_ID = "0",
  IMAGE_STATE = "2",
  LN_TARGET = "SYSTEM",
  ORIGINAL_SIZE = "256",
  READONLY = "NO",
  SAVE = "NO",
  SIZE = "256",
  SOURCE = "/var/lib/one//datastores/1/25544a9df59520398cc22759b1cc40c8",
  TARGET = "vda",
  TM_MAD = "ssh",
  TYPE = "FILE" ]
```

What OpenNebula version are you using?

**Comment by dfelsch** [comment]
I'm using 5.12.0.3-1.ce. The only difference in the templates that I see is that in my case, the image is owned by `oneadmin` and the user that instantiated the VM only has USE rights on the image.

**Comment by al3xhh** [comment]
I'm using the beta version, is it possible that you update to it and let me know if the bug happens to you? We have made some changes in this updateconf functionality in the new version, so maybe the issue is solved.

**Comment by dfelsch** [comment]
I observed this on my production setup that I won't update to a beta. But I can try to reproduce the bug on a test setup and update that to see if it disappears.

**Comment by al3xhh** [comment]
Yes yes, on a production not, but if you can create a test environment from your current production environment it would be awesome. Many thanks!!

**Comment by dfelsch** [comment]
No, the bug is still there, tested it on current master (b5e703f2a7191364c56dd0468beaeaead01ace06)
![image](https://user-images.githubusercontent.com/17570892/110953858-1baae280-8348-11eb-87c0-d9dc5767dece.png)

**Comment by al3xhh** [comment]
I also tried with master and I wasn't able to reproduce it. Could you please provide:

- A concise description of all the steps you are following.
- Did you use `oned.conf` from Master too?
- The output of `onevm show <VM_ID> -x`.
- The output of `oneuser show <USER_ID> -x` that is performing the update.

**Comment by dfelsch** [comment]
Super detailed steps to reproduce:
1) Fresh clean install of one
1) Log in as oneadmin
1) Create new image
    * Name: `Empty`
    * Type: OS
    * Persistent: No
    * Select "Empty disk image" of 128 MB
    * Bus: virtio
    * Format: qcow2
1) Give `Use` right on image to `Other`
1) Create new template
    * Name: `Test`
    * Select `KVM`
    * 1 GB RAM
    * 1 CPU / 1 VCPU
    * As storage select image `Empty`
1) Give `Use` right on template to `Other`
1) Edit group `users` and allow using the User View (for convenience)
1) Create new user `testuser` in group `users`
1) Create dummy host (KVM, hostname `localhost`)
    * goes instantly to ERROR state because there's no node package installed, does not matter
1) Log in as testuser
1) Instantiate template `Test`
1) As oneadmin deploy the VM and recover->success it until it's RUNNING, then `shutdown (hard)` and again recover->success
    * only to have the VM in POWEROFF state
1) As testuser `Update Configuration` and change the CPU model (any other modification also works)
1) Error occurs

`oned.conf` is from master (sha256 f334d8cce4b4804a70e57fa72e622d025b7ee21c9227c321adb4fd02281dec2f)

[onevm_show.txt](https://github.com/OpenNebula/one/files/6130843/onevm_show.txt)
[oneuser_show.txt](https://github.com/OpenNebula/one/files/6130844/oneuser_show.txt)

**Comment by al3xhh** [comment]
Thank you a lot!!!! I will try with those steps. Just to double check, if you try `onevm updateconf` does it work?

**Comment by dfelsch** [comment]
The CLI command `onevm updateconf 0 --user testuser` works. Seems to be an issue with Sunstone.

**Comment by al3xhh** [comment]
Thank you for all the tests @dfelsch, we are going to fix Sunstone part.

Best,
Álex.

**Comment by al3xhh** [comment]
PRs to merge in master:

- [code](https://github.com/OpenNebula/one-ee/pull/977)
- [docs](https://github.com/OpenNebula/docs/pull/1515)

**Comment by OpenNebulaSupport** [comment]
Reopening it as it is not fixed in 5.12.8 version.

**Comment by tinova** [comment]
fixed in ba9983bd7dab47267ed77731fcd0d7ecdb1e9c04 and 7814dc597a3d16cb8274c5d4a5992cf2f7a9eac8
